rpc/client.test.ts

```diff
@@ -97,24 +89,26 @@ Deno.test({
       name: "subscribe general error",
       async fn() {
         const { client, emitEvent } = createMockClient()
-        const message = { id: client.providerRef.nextId() } as msg.EgressMessage
-        const response = client.subscribe(message, (event) => A.assertInstanceOf(event, Error))
+        const message = { id: 0 } as msg.EgressMessage
+        const pending = client
+          .subscriptionFactory()(null!, null!, null!, ({ end }) => () => end())
         emitEvent(new ProviderSendError(null!, message))
-        A.assertEquals(await response, undefined)
+        A.assertEquals(await pending, undefined)
         assertClientCleanup(client)
       },
     })
 
     await t.step({
       name: "subscribe error after subscribing",
+      ignore: true,
```

Comment:
This test has a pending promise because the mock doesn't handle the unsubscription call.

https://github.com/paritytech/capi/blob/db9d0a52100ecb6755627165bd85a555bbe30937/rpc/client.ts#L100-L106